plugins/modules/remote_management/redfish/ilo_redfish_info.py

```diff
@@ -0,0 +1,161 @@
+    get information back.
+  - For use with HPE iLO operations that require Redfish OEM extensions.
+options:
+  category:
+    required: true
+    description:
+      - List of categories to execute on iLO.
+    type: list
+    elements: str
+  command:
+    required: true
+    description:
+      - List of commands to execute on iLO.
+    type: list
+    elements: str
+  baseuri:
+    required: true
+    description:
+      - Base URI of iLO.
+    type: str
+  username:
+    description:
+      - User for authentication with iLO.
+    type: str
+  password:
+    description:
+      - Password for authentication with iLO.
+    type: str
+  auth_token:
+    description:
+      - Security token for authentication with iLO.
+    type: str
+    version_added: 2.3.0
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to iLO.
+    default: 10
+    type: int
+
+requirements:
+    - "python >= 3.8"
+    - "ansible >= 3.2"
+author:
+    - "Bhavya B (@bhavya06)"
+'''
+
+EXAMPLES = '''
+  - name: Get iLO Sessions
+    community.general.ilo_redfish_info:
+      category: Sessions
+      command: GetiLOSessions
+      baseuri: "{{ baseuri }}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+'''
+
+CATEGORY_COMMANDS_ALL = {
+    "Sessions": ["GetiLOSessions"]
+}
+
+CATEGORY_COMMANDS_DEFAULT = {}
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible_collections.community.general.plugins.module_utils.ilo_redfish_utils import iLORedfishUtils
+
+
+def main():
+    result = {}
+    category_list = []
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, type='list', elements='str'),
+            command=dict(required=True, type='list', elements='str'),
+            baseuri=dict(required=True),
+            username=dict(),
+            password=dict(no_log=True),
+            auth_token=dict(no_log=True),
+            timeout=dict(type='int', default=10)
+        ),
+        required_together=[
+            ('username', 'password'),
+        ],
+        required_one_of=[
+            ('username', 'auth_token'),
+        ],
+        mutually_exclusive=[
+            ('username', 'auth_token'),
+        ],
+        supports_check_mode=True
+    )
+
+    creds = {"user": module.params['username'],
+             "pswd": module.params['password'],
+             "token": module.params['auth_token']}
+
+    timeout = module.params['timeout']
+
+    root_uri = "https://" + module.params['baseuri']
+    rf_utils = iLORedfishUtils(creds, root_uri, timeout, module)
+
+    # Build Category list
+    if "all" in module.params['category']:
+        for entry in CATEGORY_COMMANDS_ALL:
+            category_list.append(entry)
+    else:
+        # one or more categories specified
+        category_list = module.params['category']
+
+    for category in category_list:
+        command_list = []
```

Comment:
You are basically throwing away this list for previous categories.

Reply:
Code corrected. Please recheck